### PR TITLE
modify thrift's serialization so that it can fit in v0.95

### PR DESCRIPTION
**Pull request by ykwd**

### CODE REVIEW

**Comment by qinzuoyan** [comment]
now compatible with v0.95 for thrift?

**Comment by ykwd** [comment]
@qinzuoyan  yes. You may notice that  I add some extra steps like "writeStructBegin", "writeFieldStop", etc. They are empty functions for binary protocol, but are really meaningful and necessary for json protocol.

File: include/dsn/js/dsn_types.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  if (this.code !== null && this.code !== undefined) {
+    output.writeFieldBegin('code', Thrift.Type.STRING, 1);
+    output.writeString(this.code);
+    output.writeFieldEnd();
+  }
+  output.writeFieldStop();
+  output.writeStructEnd();
+  return;
+};
+
```
Comment (imzhenyu) [inline include/dsn/js/dsn_types.js:113]: Thanks @ykwd . How about the other built-in data structures such as rpc_address and gpid, both are widely used in many later data structures.

**Comment by ykwd** [comment]
@imzhenyu  rpcaddress and gpid can be safely transformed within rdsn now, both in binary form and json form, as they are treated as integers. And chang said that they are not used in js.

**Pull request merged**